src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -79,6 +93,35 @@ internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
             this.enableW3CHeaders = enableW3CHeaders;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HostingDiagnosticListener"/> class.
+        /// </summary>
+        /// <param name="configuration"><see cref="TelemetryConfiguration"/> as a settings source.</param>
+        /// <param name="client"><see cref="TelemetryClient"/> to post traces to.</param>
+        /// <param name="applicationIdProvider">Provider for resolving application Id to be used in multiple instruemntation keys scenarios.</param>
+        /// <param name="injectResponseHeaders">Flag that indicates that response headers should be injected.</param>
+        /// <param name="trackExceptions">Flag that indicates that exceptions should be tracked.</param>
+        /// <param name="enableW3CHeaders">Flag that indicates that W3C header parsing should be enabled.</param>
+        /// <param name="enableNewDiagnosticEvents">Flag that indicates that new diagnostic events are supported by AspNetCore</param>
+        public HostingDiagnosticListener(
```

Comment:
can this ctor() call the existing ctor() and do experimentflag, instead of fully duplicating ctor() logic here as well.

Reply:
Sure